pkg/operator/controller.go

```diff
@@ -1,484 +1,331 @@
 	}
 
-	return controller, nil
+	return reconcile.Result{}, nil
 }
 
-// eventProcessor is a long-running function that will continually
-// read and process a message on the workqueue.
-func (c *Controller) eventProcessor() {
-	for {
-		// Wait until there is a new item in the working queue
-		obj, shutdown := c.workqueue.Get()
-		if shutdown {
-			return
-		}
+func (r *TunedReconciler) reconcileResource(ctx context.Context, tuned *tunedv1.Tuned) error {
 
-		klog.V(2).Infof("got event from workqueue")
-		func() {
-			defer c.workqueue.Done(obj)
-			var workqueueKey wqKey
-			var ok bool
+	klog.V(2).Infof("sync(): Kind %s: %s/%s", tuned.Kind, tuned.Namespace, tuned.Name)
+	defaultTuned, err := r.getDefaultTuned(ctx)
 
-			if workqueueKey, ok = obj.(wqKey); !ok {
-				c.workqueue.Forget(obj)
-				klog.Errorf("expected wqKey in workqueue but got %#v", obj)
-				return
-			}
-
-			if err := c.sync(workqueueKey); err != nil {
-				requeued := c.workqueue.NumRequeues(workqueueKey)
-				// Limit retries to maxRetries.  After that, stop trying.
-				if requeued < maxRetries {
-					klog.Errorf("unable to sync(%s/%s/%s) requeued (%d): %v", workqueueKey.kind, workqueueKey.namespace, workqueueKey.name, requeued, err)
-
-					// Re-enqueue the workqueueKey.  Based on the rate limiter on the queue
-					// and the re-enqueue history, the workqueueKey will be processed later again.
-					c.workqueue.AddRateLimited(workqueueKey)
-					return
-				}
-				klog.Errorf("unable to sync(%s/%s/%s) reached max retries (%d): %v", workqueueKey.kind, workqueueKey.namespace, workqueueKey.name, maxRetries, err)
-				// Dropping the item after maxRetries unsuccessful retries.
-				c.workqueue.Forget(obj)
-				return
-			}
-			klog.V(1).Infof("event from workqueue (%s/%s/%s) successfully processed", workqueueKey.kind, workqueueKey.namespace, workqueueKey.name)
-			// Successful processing.
-			c.workqueue.Forget(obj)
-		}()
-	}
-}
-
-func (c *Controller) sync(key wqKey) error {
-	var (
-		cr           *tunedv1.Tuned
-		err, lastErr error
-	)
-	klog.V(2).Infof("sync(): Kind %s: %s/%s", key.kind, key.namespace, key.name)
-
-	if key.kind == wqKindTuned && key.name == tunedv1.TunedDefaultResourceName {
-		// default Tuned changed or a bootstrap event received
-		cr, err = c.syncTunedDefault()
-		if err != nil {
-			return fmt.Errorf("failed to sync default Tuned CR: %v", err)
-		}
-	} else {
-		cr, err = c.listers.TunedResources.Get(tunedv1.TunedDefaultResourceName)
-		if err != nil {
-			if errors.IsNotFound(err) {
-				cr, err = c.syncTunedDefault()
-				if err != nil {
-					return fmt.Errorf("failed to sync default Tuned CR: %v", err)
-				}
-			} else {
-				return fmt.Errorf("failed to get Tuned %s: %v", tunedv1.TunedDefaultResourceName, err)
-			}
-		}
+	if err != nil {
+		return fmt.Errorf("failed to get default Tuned CR: %v", err)
 	}
-	// We have the default Tuned custom resource (cr)
 
-	switch cr.Spec.ManagementState {
+	switch defaultTuned.Spec.ManagementState {
```

Comment:
@jmencak Do you know if someone is really using management state? I think most of OpenShift controllers do not use it.

Reply:
> @jmencak Do you know if someone is really using management state? I think most of OpenShift controllers do not use it.

Not sure what you're asking.  Customers using this functionality in NTO or other operators apart from NTO?  If the former, then no, I do not know.  If the latter, yes, there are OpenShift operators that implement this (dns-operator, cluster-image-registry-operator, cluster-kube-apiserver-operator, cluster-network-operator, ...).  If they do not, then they probably do not comply.